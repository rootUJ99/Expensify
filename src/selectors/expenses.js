import moment from 'moment';
//Get visibility Expenses
export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        //for 
        const createdAtMoment=moment(expense.createdAt);
        //for calander
        const startDateMatch =startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') :true
        //for calander
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true
        //for searching purpose
        const textMatch = typeof text !== 'string' || (expense.description.toLocaleLowerCase()).includes(text.toLocaleLowerCase())
        return startDateMatch && endDateMatch && textMatch
        //for select menu
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    })
}
