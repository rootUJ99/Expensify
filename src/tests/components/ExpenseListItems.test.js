import React from 'react';
import ExpenseListItem from '../../components/ExpenseListItems';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
//render expenseListItems
test('should test expenseListItems',()=>{
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});