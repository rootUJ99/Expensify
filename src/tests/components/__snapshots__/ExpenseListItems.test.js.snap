// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test expenseListItems 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      Gum
    </Link>
  </h3>
  <h4>
    $195.00
    -
    1st January 1970
  </h4>
</div>
`;
